shell/platform/darwin/ios/ios_external_texture_metal.mm

```diff
@@ -65,9 +68,123 @@
+      FML_DLOG(ERROR) << "Could not create Metal texture from pixel buffer: CVReturn " << cv_return;
+      return nullptr;
+    }
+  }
+
+  CVMetalTextureRef uv_metal_texture_raw = NULL;
+  {
+    auto cv_return =
+        CVMetalTextureCacheCreateTextureFromImage(kCFAllocatorDefault,        // allocator
+                                                  texture_cache_,             // texture cache
+                                                  pixel_buffer,               // source image
+                                                  NULL,                       // texture attributes
+                                                  MTLPixelFormatRG8Unorm,     // pixel format
+                                                  texture_size.width() / 2,   // width
+                                                  texture_size.height() / 2,  // height
+                                                  1u,                         // plane index
+                                                  &uv_metal_texture_raw       // [out] texture
+        );
+
+    if (cv_return != kCVReturnSuccess) {
+      FML_DLOG(ERROR) << "Could not create Metal texture from pixel buffer: CVReturn " << cv_return;
+      return nullptr;
+    }
+  }
+
+  fml::CFRef<CVMetalTextureRef> y_metal_texture(y_metal_texture_raw);
+
+  GrMtlTextureInfo y_skia_texture_info;
+  y_skia_texture_info.fTexture = sk_cf_obj<const void*>{
+      [reinterpret_cast<NSObject*>(CVMetalTextureGetTexture(y_metal_texture)) retain]};
+
+  GrBackendTexture y_skia_backend_texture(texture_size.width(),   // width
+                                          texture_size.height(),  // height
+                                          GrMipMapped ::kNo,      // mip-mapped
+                                          y_skia_texture_info     // texture info
+  );
+
+  fml::CFRef<CVMetalTextureRef> uv_metal_texture(uv_metal_texture_raw);
+
+  GrMtlTextureInfo uv_skia_texture_info;
+  uv_skia_texture_info.fTexture = sk_cf_obj<const void*>{
+      [reinterpret_cast<NSObject*>(CVMetalTextureGetTexture(uv_metal_texture)) retain]};
+
+  GrBackendTexture uv_skia_backend_texture(texture_size.width(),   // width
+                                           texture_size.height(),  // height
+                                           GrMipMapped ::kNo,      // mip-mapped
+                                           uv_skia_texture_info    // texture info
+  );
+  GrBackendTexture nv12TextureHandles[] = {y_skia_backend_texture, uv_skia_backend_texture};
+  SkYUVAIndex yuvaIndices[4] = {
+      SkYUVAIndex{0, SkColorChannel::kR},  // Read Y data from the red channel of the first texture
+      SkYUVAIndex{1, SkColorChannel::kR},  // Read U data from the red channel of the second texture
+      SkYUVAIndex{1,
+                  SkColorChannel::kG},  // Read V data from the green channel of the second texture
+      SkYUVAIndex{-1, SkColorChannel::kA}};  //-1 means to omit the alpha data of YUVA
```

Comment:
If we are omitting the alpha data, why is this method referring to `yuva`?

Reply:
Because some YUV formats have alpha channels (such as AYUV), skia may need to be compatible with such formats. In addition, I refer to the implementation of `SkImage::MakeFromNV12TexturesCopy`.

Reply:
And NV12 has no alpha channel.

Reply:
Should we change the name of this method then?  It's called `WrapYUVAExternalPixelBuffer`.

Reply:
good idea，it should be `WrapNV12ExternalPixelBuffer`

Reply:
done